agent/engine/docker_task_engine.go

```diff
@@ -531,6 +532,22 @@ func updateContainerMetadata(metadata *dockerapi.DockerContainerMetadata, contai
 	if len(metadata.PortBindings) != 0 && len(container.GetKnownPortBindings()) == 0 {
 		container.SetKnownPortBindings(metadata.PortBindings)
 	}
+
+	// update port mappings for service connect bridge mode.
+	// For bridge-mode ServiceConnect-enabled task, container port mappings are applied to the associated pause container
+	if task.IsServiceConnectEnabled() && task.IsNetworkModeBridge() &&
+		!container.IsInternal() && len(container.Name) > 0 {
+		pauseContainer, err := task.GetBridgeModePauseContainerForTaskContainer(container)
+		if err != nil {
+			logger.Error("Error resolving pause container for bridge mode SC container", logger.Fields{
+				field.Container: container.Name,
+				field.Error:     err,
+			})
+		} else {
+			container.SetKnownPortBindings(pauseContainer.GetKnownPortBindings())
```

Comment:
> container port mappings are applied to the associated pause container

This seems to be doing the opposite of what the comment says. Isn't this code applying the pause container's port bindings to `container` ?

Reply:
i see, updated comment makes sense, thanks!

Reply:
Thanks for asking this! Hope it makes more sense after adding example container name `~internal-ecs-pause-<$APP_CONTAINER>` and `<$APP_CONTAINER>` to the comment!